File: apps/android/feature/canvas/domain/src/main/java/com/scribblefit/feature/canvas/domain/usecase/ExecuteQuickActionUseCase.kt
package com.scribblefit.feature.canvas.domain.usecase

import com.scribblefit.feature.canvas.domain.repository.CanvasRepository

enum class QuickActionType { REPEAT_LAST, REST_DAY, RUN_5K }

class ExecuteQuickActionUseCase(private val canvasRepository: CanvasRepository) {
    suspend fun execute(type: QuickActionType) {
        val rawText = when (type) {
            QuickActionType.REPEAT_LAST -> "Repeat last workout"
            QuickActionType.REST_DAY -> "Rest day"
            QuickActionType.RUN_5K -> "Run 5k"
        }
        canvasRepository.addScribble(rawText)
    }
}

File: apps/ios/ScribbleFit/Sources/Features/Canvas/Domain/UseCases/ExecuteQuickActionUseCase.swift
import Foundation

public enum QuickActionType: String, CaseIterable, Sendable {
    case repeatLast = "Repeat last"
    case restDay = "Rest day"
    case run5k = "Run 5k"
}

public final class ExecuteQuickActionUseCase: Sendable {
    private let canvasRepository: any CanvasRepository

    public init(canvasRepository: any CanvasRepository) {
        self.canvasRepository = canvasRepository
    }

    public func execute(type: QuickActionType) async throws {
        try await canvasRepository.addScribble(rawText: type.rawValue)
    }
}
